packages/core/lib/commands/compile.js

```diff
@@ -91,7 +91,9 @@ const command = {
     }
 
     const result = await WorkflowCompile.save(config, compilationOutput);
-    await WorkflowCompile.assignNames(config, result);
+    if (config.db && config.db.enabled) {
```

Comment:
ooh! I wonder if the error from #4141 was related to this line!

Reply:
@eggplantzzz (sorry, couldn't comment on this without unresolved this conversation)

Does it make sense to move this guard to the `getDb()` function? Of course we'd have to change the interface to accept a `ConfigType` or null, executing the guard only if a config were passed.